### PR TITLE
print value and stack trace

### PR DESCRIPTION
**Pull request by piiswrong**

### CODE REVIEW

**Comment by piiswrong** [comment]
@tqchen

File: include/dmlc/logging.h [hunk 1]
```diff
@@ -14,6 +14,10 @@
 #include <stdexcept>
 #include "./base.h"
 
+#if DMLC_LOG_STACK_TRACE
+#include <execinfo.h>
```
Comment (khotilov) [inline include/dmlc/logging.h]: MinGW (at least the versions 4.9.3 and 6.2.0 that I've got on my machine) doesn't seem to have execinfo.h

File: include/dmlc/base.h [hunk 2]
```diff
@@ -37,6 +37,14 @@
 #define DMLC_LOG_CUSTOMIZE 0
 #endif
 
+/*!
+ * \brief Wheter to print stack trace for fatal error,
+ * enabled on linux when using gcc.
+ */
+#if (!defined(DMLC_LOG_STACK_TRACE) && defined(__GNUC__))
```
Comment (khotilov) [inline include/dmlc/base.h]: A check for `&& !defined(__MINGW32__)` or `&& !defined(WIN32)` might be needed. I assume that 64 bit versions have those defined too.

**Comment by tqchen** [comment]
Sorry I did not raise this up when it get merged. But it seems to be better to do everything via macro here to enforce code inlining and avoid the layer of additional inline function call. Consider instead

```c++
#define CHECK_NE(x, y)  \
     CHECK(x!=y) << x << " vs "  << y
```

**Comment by tqchen** [comment]
@piiswrong

**Comment by piiswrong** [comment]
You can't print the value of x and y with macros

**Pull request merged**